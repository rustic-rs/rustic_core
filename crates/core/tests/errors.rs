use rstest::{fixture, rstest};

use rustic_core::{ErrorKind, RusticError, Severity, Status};

#[fixture]
fn error() -> Box<RusticError> {
    RusticError::with_source(
        ErrorKind::InputOutput,
        "A file could not be read, make sure the file at `{path}` is existing and readable by the system.",
        std::io::Error::new(std::io::ErrorKind::ConnectionReset, "Networking Error"),
    )
    .attach_context("path", "/path/to/file")
    .attach_context("called", "used s3 backend")
    .attach_status(Status::Permanent)
    .attach_severity(Severity::Error)
    .attach_error_code("C001")
    .append_guidance_line("Appended guidance line")
    .prepend_guidance_line("Prepended guidance line")
    .attach_existing_issue_url("https://github.com/rustic-rs/rustic_core/issues/209")
    .ask_report()
}

#[rstest]
fn test_error_display(error: Box<RusticError>) {
    insta::assert_snapshot!(error);
}

#[rstest]
fn test_error_debug(error: Box<RusticError>) {
    insta::assert_debug_snapshot!(error);
}
