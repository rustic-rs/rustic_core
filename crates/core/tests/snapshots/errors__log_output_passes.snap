---
source: crates/core/tests/errors.rs
expression: error.to_log_output()
---
Error: Prepended guidance line
A file could not be read, make sure the file at `/path/to/file` is existing and readable by the system.
Appended guidance line (kind: related to input/output operations, code: C001): caused by: Networking Error
